s3torchconnector/src/s3torchconnector/_s3dataset_common.py

```diff
@@ -42,22 +38,23 @@ def parse_s3_uri(uri: str) -> Tuple[str, str]:
 
 def get_objects_from_uris(
     object_uris: Union[str, Iterable[str]], client: S3Client
-) -> Iterable[S3Reader]:
+) -> Iterable[S3BucketKey]:
     if isinstance(object_uris, str):
         object_uris = [object_uris]
     # TODO: We should be consistent with URIs parsing. Revise if we want to do this upfront or lazily.
     bucket_key_pairs = [parse_s3_uri(uri) for uri in object_uris]
 
-    return bucket_key_pairs_to_objects(bucket_key_pairs, client)
+    return bucket_key_pairs_to_objects(bucket_key_pairs)
```

Comment:
Nit: I'd prefer `return (S3BucketKey(bucket, key) for bucket, key in bucket_key_pairs)` here but what we've got is fine.
Or even just returning an actual list instead would be fine

Reply:
Do we want a list? Didn't we want them lazily loaded?